universum/configuration_support.py

```diff
@@ -33,8 +34,19 @@ def __init__(self,
                  pass_tag: str = '',
                  fail_tag: str = '',
                  if_env_set: str = '',
-                 extras: Optional[Dict[str, str]] = None,
-                 **kwargs) -> None:  # explicit constructor for client's type safety
+                 **kwargs) -> None:
+        """
+        All params supplied via named attributed shall be  type-checked for safety. Clients may provide custom
+        parameters in kwargs - these parameters will be stored internally in '_extras' dict and can be retrieved by
+        indexing.
+        >>> cfg = ProjectConfiguration(name='foo', command=['1', '2', '3'], _extras={'_extras': 'test', 'bool': True})
```

Comment:
The doctest is confusing because you used the same name for the dictionary to hold extra values and for the extra value itself.

Reply:
this was intentional to show that nothing breaks internally if names match

Reply:
e.g. the clients/user doesn't need to care about _extras being reserved, etc